src/server/streamableHttp.ts

```diff
@@ -0,0 +1,368 @@
+import { IncomingMessage, ServerResponse } from "node:http";
+import { Transport } from "../shared/transport.js";
+import { JSONRPCMessage, JSONRPCMessageSchema, RequestId } from "../types.js";
+import getRawBody from "raw-body";
+import contentType from "content-type";
+
+const MAXIMUM_MESSAGE_SIZE = "4mb";
+
+/**
+ * Configuration options for StreamableHTTPServerTransport
+ */
+export interface StreamableHTTPServerTransportOptions {
+  /**
+   * The session ID SHOULD be globally unique and cryptographically secure (e.g., a securely generated UUID, a JWT, or a cryptographic hash)
+   * 
+   * When sessionId is not set, the transport will be in stateless mode.
```

Comment:
Not a fan of calling this "stateless" vs. "stateful." A session isn't required to be stateful, nor does the existence of one mean that it _is_ stateful. Let's just refer to "session management" generally:

```suggestion
   * When there is no sessionId, the transport will not perform session management.
```